packages/syft/src/syft/node/server.py

```diff
@@ -35,134 +38,96 @@
 WAIT_TIME_SECONDS = 20
 
 
-def make_app(name: str, router: APIRouter) -> FastAPI:
-    app = FastAPI(
-        title=name,
-    )
+class AppSettings(BaseSettings):
+    name: str
+    node_type: NodeType = NodeType.DOMAIN
+    node_side_type: NodeSideType = NodeSideType.HIGH_SIDE
+    processes: int = 1
+    reset: bool = False
+    dev_mode: bool = False
+    enable_warnings: bool = False
+    in_memory_workers: bool = True
+    queue_port: int | None = None
+    create_producer: bool = False
+    n_consumers: int = 0
+    association_request_auto_approval: bool = False
+    background_tasks: bool = False
+
+    model_config = SettingsConfigDict(env_prefix="SYFT_", env_parse_none_str="None")
+
+
+def app_factory() -> FastAPI:
+    settings = AppSettings()
+
+    worker_classes = {
```

Comment:
nit: we define this worker_classes dict a few times in the codebase, can we reuse?

Reply:
Will address this in a separate PR